src/Bicep.Core/FileSystem/PathHelper.cs

```diff
@@ -137,6 +139,27 @@ public static Uri FilePathToFileUrl(string filePath)
             return uriBuilder.Uri;
         }
 
+        public static string? TryGetUsingPath(UsingDeclarationSyntax? usingDeclarationSyntax, out DiagnosticBuilder.ErrorBuilderDelegate? failureBuilder)
```

Comment:
The caller of this method will need to do one null check on a success, and two on failure. Those checks could be skipped if a slightly different signature and a couple [`NotNullWhen` annotations](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/nullable-analysis#conditional-post-conditions-notnullwhen-maybenullwhen-and-notnullifnotnull) were used.

```suggestion
        public static bool TryGetUsingPath(UsingDeclarationSyntax? usingDeclarationSyntax, [NotNullWhen(true)] out string path, [NotNullWhen(false)] out DiagnosticBuilder.ErrorBuilderDelegate? failureBuilder)
```